src/luarocks/fs/lua.lua

```diff
@@ -785,6 +770,40 @@ function fs_lua.get_permissions(file)
    return posix.stat(file, "mode")
 end
 
+--- Create a temporary directory.
+-- @param name string: name pattern to use for avoiding conflicts
+-- when creating temporary directory.
+-- @return string or (nil, string): name of temporary directory or (nil, error message) on failure.
+function fs_lua.make_temp_dir(name)
+   assert(type(name) == "string")
+   name = dir.normalize(name)
+
+   local template = (os.getenv("TMPDIR") or "/tmp") .. "/luarocks_" .. name:gsub(dir.separator, "_") .. "-XXXXXX"
+   local temp_dir, err = posix.mkdtemp(template)
+   if temp_dir then
+      return temp_dir
+   else
+      return nil, err
+   end
+end
+
+--- Create and return the name of a temporary file.
+-- @return string or (nil, string): name of temporary file or (nil, error message) on failure.
+function fs_lua.tmpname()
```

Comment:
When I wrote this I had assumed from the name that os.tmpname() just used tmpnam(), but actually looking at it the PUC lua source it looks like it will also use mkstemp where it's available.
This is probably not necessary then, the only difference would be that it would respect $TMPDIR where the os.tmpname() just hardcodes "/tmp"